examples/mouse_drag_callback.py

```diff
@@ -0,0 +1,45 @@
+"""
+Example updating the status bar with line profile info while dragging
+lines around in a shapes layer.
+"""
+
+from skimage import data
+from skimage import measure
+import numpy as np
+import napari
+
+
+def profile_lines(image, shape_layer):
+    profile_data = []
+    for line in shape_layer.data:
+        profile_data.append(
+            measure.profile_line(image, line[0], line[1]).mean()
+        )
+    msg = ('profile means: ['
+            + ', '.join([f'{d:.2f}' for d in profile_data])
+            + ']')
+    shape_layer.status = msg
+
+
+
+with napari.gui_qt():
+    np.random.seed(1)
+    viewer = napari.Viewer()
+    blobs = data.binary_blobs(length=512, volume_fraction=0.1, n_dim=2)
+    viewer.add_image(blobs, name='blobs')
+    line1 = np.array([[11, 13], [111, 113]])
+    line2 = np.array([[200, 200], [400, 300]])
+    lines = [line1, line2]
+    shapes_layer = viewer.add_shapes(lines, shape_type='line',
+            edge_width=5, edge_color='coral', face_color='royalblue')
+    shapes_layer.mode = 'select'
+
+    @shapes_layer.mouse_drag_callbacks.append
```

Comment:
Curious why not define the callback at the viewer level and then work with the viewer object to get both the `Shape` and `Image` layers - we might have to use layer names / order to grab them though which isn't so nice as these are both changeable ...

Worth considering as we think about the best api's here